compatibility_lib/compatibility_lib/package_crawler_static.py

```diff
@@ -219,9 +219,11 @@ def _get_function_info(functions):
 
 
 def _get_args(node):
-    """returns a list of args"""
+    """returns a list of non-default args, ignoring args with defaults"""
     args = []
-    for arg in node.args:
+    num_args = len(node.args) - len(node.defaults)
```

Comment:
Python argument lists can be pretty complex e.g.
def baz(foo, bar=3, *args, key=5): pass

You might want to write a bunch of tests to make sure that you handle all of the edge cases.

Reply:
Oh nice catch - I forgot about that way of passing args.

Quick question about *args and **kwargs though: would a signature like this ever appear?
`foo(a, *args, c): pass`

Reply:
I don't know how often that appears in practice. I like to do foo(a, *, b, c, d) because I'm a fan of keyword arguments ;-)

Reply:
i'll add more test cases!

Reply:
done